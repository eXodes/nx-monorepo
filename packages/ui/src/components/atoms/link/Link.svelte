<script lang="ts">
import { ExternalLink, Icon } from "svelte-hero-icons";

export let href: string;
export let target = "_self";
export let external = false;
</script>

<a
  class="text-brand-600 hover:text-brand-500 dark:text-brand-400 dark:hover:text-brand-500 inline-flex hover:underline"
  href={href}
  target={external ? "_blank" : target}
  data-testid="link"
>
  <slot />
  {#if external}
    <span class="my-auto ml-0.5 inline">
      <Icon src={ExternalLink} class="h-4 w-4" />
    </span>
  {/if}
</a>
