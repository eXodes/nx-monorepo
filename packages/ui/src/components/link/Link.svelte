<script lang="ts">
  export let href: string;
  export let target = "_self";
</script>

<a class="hover:underline" href={href} target={target} data-testid="link">
  <slot />
</a>
