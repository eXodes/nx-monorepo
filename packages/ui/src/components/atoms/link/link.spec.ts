import Link from "./Link.svelte";
import { render } from "@testing-library/svelte";

describe("Link", () => {
  it("renders", () => {
    const { container } = render(Link, {
      props: {
        href: "https://link.test",
      },
    });

    expect(container).toMatchSnapshot();
  });

  it("renders with external icon", () => {
    const { container } = render(Link, {
      props: {
        href: "https://link.test",
        external: true,
      },
    });

    expect(container).toMatchSnapshot();
  });

  it("should have href props defined", async () => {
    const { getByTestId } = render(Link, {
      props: {
        href: "https://link.test",
      },
    });

    const component = getByTestId("link");

    expect(component.getAttribute("href")).toContain("https://link.test");
  });

  it("should open external link in a new tab", async () => {
    const { getByTestId } = render(Link, {
      props: {
        href: "https://link.test",
        external: true,
      },
    });

    const component = getByTestId("link");

    expect(component.getAttribute("target")).toContain("_blank");
  });
});
