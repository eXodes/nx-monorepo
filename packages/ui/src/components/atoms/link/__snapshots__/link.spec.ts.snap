// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
<body>
  <div>
    <a
      class="text-brand-600 hover:text-brand-500 dark:text-brand-400 dark:hover:text-brand-500 inline-flex hover:underline"
      data-testid="link"
      href="https://link.test"
      target="_self"
    >
       
    </a>
  </div>
</body>
`;

exports[`Link renders with external icon 1`] = `
<body>
  <div>
    <a
      class="text-brand-600 hover:text-brand-500 dark:text-brand-400 dark:hover:text-brand-500 inline-flex hover:underline"
      data-testid="link"
      href="https://link.test"
      target="_blank"
    >
       
      <span
        class="my-auto ml-0.5 inline"
      >
        <svg
          aria-hidden="true"
          class="h-4 w-4"
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
    </a>
  </div>
</body>
`;
